en/mavgen_c/README.md

```diff
@@ -59,16 +60,166 @@ When compiling the project, we recommend that you specify the top-level output d
+The C MAVLink library utilizes a "channel" metaphor to allow for simultaneous processing of multiple, independent MAVLink streams in the same program.
+All receiving and transmitting functions provided by this library require a channel, and it is important to use the correct channel for each operation.
+
+By default up to 16 channels may be defined on Windows, Linux and macOS, and up to 4 channels may be define on other systems.
+Systems can specify a different maximum number of channels/comms buffers using `MAVLINK_COMM_NUM_BUFFERS` (for example, this might be reduced to 1 if running MAVLink on very memory constrained hardware).
 
 If only one MAVLink stream exists, channel 0 should be used by specifying the `MAVLINK_COMM_0` constant.
 
+
 ## Receiving
 
-MAVLink reception is then done using `mavlink_helpers.h:mavlink_parse_char()`.
+MAVLink reception/decoding is done in a number of phases: 
+1. Parse the incoming stream into objects representing each packet (`mavlink_message_t`).
+1. Decode the MAVLink message contained in the packet payload into a C struct (that has fields mapping the original XML definition).
+
+These steps are demonstrated below.
+
+### Parsing Packets
+
+The `mavlink_parse_char(...)` convenience function ([mavlink_helpers.h](https://github.com/mavlink/c_library_v2/blob/master/mavlink_helpers.h)) is used to parse incoming MAVLink data.
+The function parses the data one byte at a time, returning 0 (`MAVLINK_FRAMING_INCOMPLETE`) as it progresses, and 1 (`MAVLINK_FRAMING_OK`) on successful decoding of a packet.
+The `r_mavlink_status` parameter is updated with the channel status/errors as decoding progresses (you can check `mavlink_status_t.msg_received` to get the current byte's decode status/error and `parse_statemavlink_status_t` for the current parse state).
+On successful decoding of a packet, the `r_message` argument is populated with an object representing the decoded packet.
+
+The function prototype and parameters are shown below:
+```c
+MAVLINK_HELPER uint8_t mavlink_parse_char(uint8_t chan, uint8_t c, mavlink_message_t* r_message, mavlink_status_t* r_mavlink_status)
+```
+
+Parameters:
+- `uint8_t chan`: ID of the current channel.
+- `uint8_t c`: The char to parse.
+- [`mavlink_message_t*`](https://github.com/ArduPilot/pymavlink/blob/c0f5bb2695a677721aa7bb8f20be40ba8274c3d4/generator/C/include_v2.0/mavlink_types.h#L108) `r_message`: On success, the decoded message. NULL if the message couldn't be decoded.
+- [`mavlink_status_t*`](https://github.com/ArduPilot/pymavlink/blob/c0f5bb2695a677721aa7bb8f20be40ba8274c3d4/generator/C/include_v2.0/mavlink_types.h#L217) `r_mavlink_status`: The channel statistics, including information about the current parse state.
+
+Returns: `0` if packet decoded. `1` if packet decoded.
+
+
+The code fragment below shows the typical use of this function when reading data from a serial port (`serial`):
+```c
+#include <mavlink.h>
```

Comment:
I think this include is wrong, shouldn't it be `#include <common/mavlink.h>`, or `#include <standard/mavlink.h>`?

Reply:
Yes. Thanks!